src/v/cluster/cluster_utils.h

```diff
@@ -293,6 +293,27 @@ inline std::vector<model::broker_shard> subtract_replica_sets_by_node_id(
       });
     return ret;
 }
+
+inline std::vector<model::broker_shard> subtract_replica_sets_by_node_id(
+  const std::vector<model::broker_shard>& lhs,
+  const std::vector<model::node_id>& rhs) {
+    std::vector<model::broker_shard> ret;
+    std::copy_if(
+      lhs.begin(),
+      lhs.end(),
+      std::back_inserter(ret),
+      [&rhs](const model::broker_shard& lhs_bs) {
+          return std::find_if(
+                   rhs.begin(),
+                   rhs.end(),
+                   [&lhs_bs](const model::node_id& rhs_node) {
+                       return rhs_node == lhs_bs.node_id;
+                   })
+                 == rhs.end();
+      });
+    return ret;
+}
+
```

Comment:
nit: this can be a function template accepting predicate, it would prevent code duplication